apps/dashboard/src/components/conditions-editor/value-editor.tsx

```diff
@@ -32,26 +37,32 @@ export const ValueEditor = React.memo((props: ValueEditorProps) => {
     });
 
     return (
-      <div className="flex items-center gap-1">
-        {editors[0]}
-        <span className="text-foreground-600 text-paragraph-xs">and</span>
-        {editors[1]}
+      <div className="flex flex-col gap-1">
+        <div className="flex items-start gap-1">
+          {editors[0]}
+          <span className="text-foreground-600 text-paragraph-xs mt-1.5">and</span>
+          {editors[1]}
+        </div>
+        {error && <span className="text-destructive text-xs">{error?.message}</span>}
       </div>
     );
   }
 
   return (
-    <InputRoot size="2xs" className="w-40">
-      <InputWrapper className="h-7">
-        <ControlInput
-          multiline={false}
-          indentWithTab={false}
-          placeholder="value"
-          value={value ?? ''}
-          onChange={handleOnChange}
-          variables={variables}
-        />
-      </InputWrapper>
-    </InputRoot>
+    <div className="flex flex-col gap-1">
+      <InputRoot size="2xs" className="w-40" hasError={!!error}>
+        <InputWrapper className="h-7">
+          <ControlInput
+            multiline={false}
+            indentWithTab={false}
+            placeholder="value"
+            value={value ?? ''}
+            onChange={handleOnChange}
+            variables={variables}
+          />
+        </InputWrapper>
+      </InputRoot>
+      {error && <span className="text-destructive text-xs">{error?.message}</span>}
```

Comment:
Shouldn't we use the <FormMessage component instead?

Reply:
Not really because the particular editor fields are not connected to the form, but the whole editor is treated as a "single" input.